src/main/java/com/czertainly/core/service/cmp/impl/CmpServiceImpl.java

```diff
@@ -0,0 +1,344 @@
+                return;
+            }
+            raProfile = cmpProfile.getRaProfile();
+            if (raProfile == null) {
+                return;
+            }
+            issueAttributes = attributeEngine.getRequestObjectDataAttributesContent(cmpProfile.getRaProfile().getAuthorityInstanceReference().getConnectorUuid(), AttributeOperation.CERTIFICATE_ISSUE, Resource.CMP_PROFILE, cmpProfile.getUuid());
+            revokeAttributes = attributeEngine.getRequestObjectDataAttributesContent(cmpProfile.getRaProfile().getAuthorityInstanceReference().getConnectorUuid(), AttributeOperation.CERTIFICATE_REVOKE, Resource.CMP_PROFILE, cmpProfile.getUuid());
+        }
+
+        Certificate cmpCaCertificate = cmpProfile.getSigningCertificate();
+        if (cmpCaCertificate == null) {
+            throw new CmpConfigurationException(PKIFailureInfo.systemFailure,
+                    "PN="+profileName+" | CMP profile does not have any associated CA certificate");
+        }
+
+        try {
+            this.recipient = CertificateUtil.parseCertificate(cmpCaCertificate.getCertificateContent().getContent());
+        } catch (CertificateException e) {
+            // This should not occur
+            throw new CmpConfigurationException(PKIFailureInfo.systemFailure,
+                    "PN="+profileName+" | Error converting the certificate to x509 object");
+        }
+        try {
+            this.caCertificateChain = loadCertificateChain(cmpCaCertificate);
+        } catch (NotFoundException e) {
+            throw new CmpConfigurationException(PKIFailureInfo.systemFailure,
+                    "PN="+profileName+" | Failed to load certificate chain of CMP profile CA certificate");
+        }
+        LOG.debug("PN={} | CMP service initialized: isRaProfileBased: {}, raProfile: {}, cmpProfile: {}", profileName, raProfileBased, raProfile, cmpProfile);
+    }
+
+    private void validateProfile(String incomingProfileName) throws CmpConfigurationException {
+        validateCmpProfile(incomingProfileName);
+        validateRaProfile(incomingProfileName);
+    }
+
+    private void validateCmpProfile(String incomingProfileName) throws CmpConfigurationException {
+        if (cmpProfile == null) {
+            throw new CmpConfigurationException(PKIFailureInfo.systemFailure,
+                    "PN="+incomingProfileName+" | Requested CMP Profile not found");
+        }
+        if (!cmpProfile.getEnabled()) {
+            throw new CmpConfigurationException(PKIFailureInfo.systemFailure,
+                    "PN="+incomingProfileName+" | CMP Profile is not enabled");
+        }
+        if (cmpProfile.getSigningCertificate() == null) {
+            throw new CmpConfigurationException(PKIFailureInfo.systemFailure,
+                    "PN="+incomingProfileName+" | CMP Profile does not have any associated CA certificate for signature");
+        }
+        if (!CertificateUtil.isCertificateCmpAcceptable(cmpProfile.getSigningCertificate())) {

```

Comment:
This is already checked in the `CmpProfileServiceImpl`. You cannot create CMP Profile with invalid signing certificate in case the protection of the response is set to `signature`.

Reply:
DONE, moved/merged together (from init into validateCmpProfile method)